pkg/controller/backup_controller.go

```diff
@@ -269,16 +273,22 @@ func (controller *backupController) processBackup(key string) error {
 
 	logContext.Debug("Running backup")
 	// execution & upload of backup
+	backupName := kubeutil.NamespaceAndName(backup)
+	backupScheduleName := getBackupScheduleName(backup)
+	controller.metrics.RegisterBackupAttempt(backupName, backupScheduleName)
+
 	if err := controller.runBackup(backup, controller.bucket); err != nil {
 		logContext.WithError(err).Error("backup failed")
 		backup.Status.Phase = api.BackupPhaseFailed
+		controller.metrics.RegisterBackupFailed(backupName, backupScheduleName)
```

Comment:
I think the `RegisterBackupSuccess` case may need to be moved into an `else` clause here, such that a backup doesn't register as failed and succeeded.

I tried to run a backup using a GCP serviceaccount that didn't have access to the bucket selected, and my metrics stated that the backup failed and succeeded.

```
ERRO[0436] Error uploading log file                      bucket=<BUCKET_REDACTED> error="rpc error: code = Unknown desc = googleapi: Error 403: <ACCOUNT_REDACTED>.iam.gserviceaccount.com does not have storage.objects.create access to <BUCKET_REDACTED>/default-backup/default-backup-logs.gz., forbidden" key=default-backup/default-backup-logs.gz logSource="pkg/cloudprovider/backup_service.go:148"
```

`ark backup get` shows the backup as failed.

```
% ./_output/bin/linux/amd64/ark backup get
NAME             STATUS    CREATED                         EXPIRES   SELECTOR
default-backup   Failed    2018-06-12 09:52:38 -0400 EDT   29d       <none>
```

These are the metrics:

```
ark_server_backup_attempt_total{backupName="heptio-ark/default-backup",schedule="UNSCHEDULED"} 1
ark_server_backup_failed_total{backupName="heptio-ark/default-backup",schedule="UNSCHEDULED"} 1
ark_server_backup_size_bytes{backupName="heptio-ark/default-backup",schedule="UNSCHEDULED"} 4865
ark_server_backup_success_total{backupName="heptio-ark/default-backup",schedule="UNSCHEDULED"} 1
```

Reply:
@nrb thanks for running the test for me and catching this bug! Pushed a change with the fix.